flow/project.py

```diff
@@ -234,54 +246,31 @@ def __str__(self):
         return "{}({})".format(self.name, self.job)
 
     def __repr__(self):
-        return "{type}(name='{name}', job='{job}', cmd={cmd}, directives={directives})".format(
-            type=type(self).__name__,
-            name=self.name,
-            job=str(self.job),
-            cmd=repr(self.cmd),
-            directives=self.directives)
-
-    def get_id(self, index=0):
-        "Return a name, which identifies this job-operation."
-        project = self.job._project
-
-        # The full name is designed to be truly unique for each job-operation.
-        full_name = '{}%{}%{}%{}'.format(
-            project.root_directory(), self.job.get_id(), self.name, index)
-
-        # The job_op_id is a hash computed from the unique full name.
-        job_op_id = calc_id(full_name)
-
-        # The actual job id is then constructed from a readable part and the job_op_id,
-        # ensuring that the job-op is still somewhat identifiable, but guarantueed to
-        # be unique. The readable name is based on the project id, job id, operation name,
-        # and the index number. All names and the id itself are restricted in length
-        # to guarantuee that the id does not get too long.
-        max_len = self.MAX_LEN_ID - len(job_op_id)
-        if max_len < len(job_op_id):
-            raise ValueError("Value for MAX_LEN_ID is too small ({}).".format(self.MAX_LEN_ID))
-
-        readable_name = '{}/{}/{}/{:04d}/'.format(
-            str(project)[:12], str(self.job)[:8], self.name[:12], index)[:max_len]
-
-        # By appending the unique job_op_id, we ensure that each id is truly unique.
-        return readable_name + job_op_id
+        return "{type}(name='{name}', operations='{operations}', " \
+               "job='{job}', cmd={cmd}, directives={directives})".format(
+                   type=type(self).__name__,
+                   name=self.name,
+                   operations=self.operations.keys(),
+                   job=str(self.job),
+                   cmd=repr(self._run_cmd),
+                   directives=self.directives)
 
     def __hash__(self):
-        return int(sha1(self.get_id().encode('utf-8')).hexdigest(), 16)
+        return int(sha1(self.id.encode('utf-8')).hexdigest(), 16)
 
     def __eq__(self, other):
-        return self.get_id() == other.get_id()
+        return self.id == other.id
 
     def set_status(self, value):
         "Store the operation's status."
         self.job._project.document.setdefault('_status', dict())
-        self.job._project.document._status[self.get_id()] = int(value)
+        self.job._project.document._status[self.id] = int(value)
```

Comment:
I believe one could make this a one-liner, because `setdefault` returns the `dict`.